File: Java/src/jakopec/t3nasljedivanje/Racun.java
package jakopec.t3nasljedivanje;

public class Racun extends Dokument {

    private String kupac;

    public Racun() {
        super();
    }

    public Racun(int sifra, String brojDokumenta, double iznos, String kupac) {
        super(sifra, brojDokumenta, iznos,null);
        this.kupac = kupac;
    }

    public String getKupac() {
        return kupac;
    }

    public void setKupac(String kupac) {
        this.kupac = kupac;
    }

    @Override
    public String toString() {
        return "Racun{" +
                "kupac='" + kupac + '\'' +
                '}' + " koji nasljeđuje " + super.toString();
    }
}

File: Kotlin/src/main/kotlin/t3nasljedivanje/Racun.kt
package t3nasljedivanje

/**
 * @author Matija Skrletović
 **/

class Racun(sifra: Int, brojDokumenta: String, iznos: Double, val kupac: String) : Dokument(sifra, brojDokumenta, iznos) {

    override fun toString(): String {
        return "Racun{kupac='$kupac'} koji nasljeđuje ${super.toString()}"
    }
}